lib/ansible/module_utils/basic.py

```diff
@@ -1591,6 +1591,13 @@ def _check_argument_values(self, spec=None, param=None):
             choices = v.get('choices', None)
             if choices is None:
                 continue
+            if BOOLEANS.intersection(choices):
```

Comment:
This feels like something that would be better handled as a sanity test, it's targeted at developers but it's going to be seen by users

Reply:
Good point, thanks! I'll refactor this PR to have two parts:

- A sanity check to discourage module developers to include boolean-like choices in string fields
- A message to the user discouraging using boolean-like choices, even if the module supports it